packages/amplify-frontend-javascript/lib/frontend-config-creator.js

```diff
@@ -274,29 +277,35 @@ function getAWSExportsObject(resources) {
 }
 
 async function getCurrentAWSExports(context) {
-  const { amplify } = context;
-  const projectPath = context.exeInfo ? context.exeInfo.localEnvInfo.projectPath : amplify.getEnvInfo().projectPath;
-  const projectConfig = context.exeInfo ? context.exeInfo.projectConfig[constants.Label] : amplify.getProjectConfig()[constants.Label];
-  const frontendConfig = projectConfig.config;
+  const { amplify, exeInfo } = context;
+  const projectPath = exeInfo?.localEnvInfo?.projectPath || amplify.getEnvInfo().projectPath;
+  const { config: frontendConfig } = exeInfo?.projectConfig?.[constants.Label] || amplify.getProjectConfig()[constants.Label];
   const srcDirPath = path.join(projectPath, frontendConfig.SourceDir);
 
   const targetFilePath = path.join(srcDirPath, constants.exportsFilename);
   let awsExports = {};
 
   if (fs.existsSync(targetFilePath)) {
-    // if packaged, we can't load an ES6 module because pkg doesn't support it yet
-    // eslint-disable-next-line spellcheck/spell-checker
-    const es5export = 'module.exports = {default: awsmobile};\n';
-    // eslint-disable-next-line spellcheck/spell-checker
-    const es6export = 'export default awsmobile;\n';
-
     const fileContents = fs.readFileSync(targetFilePath, 'utf-8');
-    fs.writeFileSync(targetFilePath, fileContents.replace(es6export, es5export));
-    // eslint-disable-next-line global-require, import/no-dynamic-require
-    awsExports = require(targetFilePath).default;
-    fs.writeFileSync(targetFilePath, fileContents);
+    try {
+      // transpile the file contents to CommonJS
+      const { code } = babel.transformSync(fileContents, {
+        plugins: [babelTransformEsmToCjs],
+      });
+      const m = new Module();
+      m._compile(code, 'aws-exports.js');
+      // add paths to the module to account for node_module imports in aws-exports.js (should there be any)
+      m.paths = [projectPath];
+      // the transpiled result will contain `exports.default`
+      awsExports = m.exports?.default || m.exports;
+    } catch (error) {
+      throw new Error('Unable to parse aws-exports.js. Has this file been modified?');
```

Comment:
Ideally update the thrown errors to be more specific once https://github.com/aws-amplify/amplify-cli/pull/10933 is merged.

Reply:
be more specific as in something like this?

```ts
class AwsExportsParseError extends Error {}
```